smart_card_connector_app/src/testing_smart_card_simulation.h

```diff
@@ -67,52 +72,62 @@ class TestingSmartCardSimulation final {
     std::vector<uint8_t> next_bulk_transfer_reply;
   };
 
-  void PostFakeJsResult(RequestId request_id, Value result);
-  void PostFakeJsError(RequestId request_id, const std::string& error);
-  void PostResponseMessage(ResponseMessageData response_data);
-
-  DeviceState* FindDeviceStateByIdLocked(int64_t device_id);
-  DeviceState* FindDeviceStateByIdAndHandleLocked(int64_t device_id,
-                                                  int64_t device_handle);
-
-  void OnListDevicesCalled(RequestId request_id);
-  void OnGetConfigurationsCalled(int64_t device_id, RequestId request_id);
-  void OnOpenDeviceHandleCalled(int64_t device_id, RequestId request_id);
-  void OnCloseDeviceHandleCalled(int64_t device_id,
-                                 int64_t device_handle,
-                                 RequestId request_id);
-  void OnClaimInterfaceCalled(int64_t device_id,
-                              int64_t device_handle,
-                              int64_t interface_number,
-                              RequestId request_id);
-  void OnReleaseInterfaceCalled(int64_t device_id,
-                                int64_t device_handle,
-                                int64_t interface_number,
-                                RequestId request_id);
-  void OnControlTransferCalled(int64_t device_id,
-                               int64_t device_handle,
-                               LibusbJsControlTransferParameters params,
-                               RequestId request_id);
-  void OnBulkTransferCalled(int64_t device_id,
-                            int64_t device_handle,
-                            LibusbJsGenericTransferParameters params,
-                            RequestId request_id);
-  void OnInterruptTransferCalled(int64_t device_id,
-                                 int64_t device_handle,
-                                 LibusbJsGenericTransferParameters params,
-                                 RequestId request_id);
-
-  optional<LibusbJsTransferResult> HandleOutputBulkTransferLocked(
-      const std::vector<uint8_t>& data_to_send,
-      DeviceState& device_state);
-  optional<LibusbJsTransferResult> HandleInputBulkTransferLocked(
-      int64_t length_to_receive,
-      DeviceState& device_state);
+  // Helper class that provides thread-safe operations with reader states.
+  class Core final {
```

Comment:
I think `Core` is a bad name - it doesn't really tell me anything about the purpose of this class. Maybe something like `DeviceStateHelper`? (Note that I have not yet read any code of the class as of now, so I'm just guessing)

Reply:
I see. Maybe `ThreadSafeHandler`?

Reply:
Sounds good